CppCoreGuidelines.md

```diff
@@ -19397,8 +19399,10 @@ An implementation of this profile shall recognize the following patterns in sour
 
 Type safety profile summary:
 
-* <a name="Pro-type-reinterpretcast"></a>Type.1: Don't use `reinterpret_cast`:
-A strict version of [Avoid casts](#Res-casts) and [prefer named casts](#Res-casts-named).
+* <a name="Pro-type-avoidcasts"></a>Type.1: [Avoid casts](#Res-casts):  
+<a name="Pro-type-reinterpretcast">a. </a>Don't use `reinterpret_cast`; A strict version of [Avoid casts](#Res-casts) and [prefer named casts](#Res-casts-named).  
+<a name="Pro-type-arithmeticcast">b. </a>Don't a `static_cast` for arithmetic types; A strict version of [Avoid casts](#Res-casts) and [prefer named casts](#Res-casts-named).  
+<a name="Pro-type-unnecessarycast">c. </a>Don't use pointer unnecessary pointer casts; A strict version of [Avoid casts](#Res-casts)  
```

Comment:
Suggestion: "<a name="Pro-type-identity-cast">c.<a/> Don't use pointer casts where the source and the target types are the same;"

This suggestion is based on the discussion from last review when I asked "what is unnecessary cast" and I was told that you had in mind "casts where the source and target types are the same."

People have complained in the past that the rules are too imprecise or unimplementable,  I want to ensure that as we add new rule or we refine existing rules, we are making them precise and definitely implementable.